lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,565 @@
+    return args_class(**kwargs) if not args else args_class(*args, **kwargs)
+
+
+# %%
+# load the pl extension module we want to use. This will import all necessary callbacks.
+module_hub_mock("finetuningscheduler")
+# set notebook-level variables
+AVAIL_GPUS = min(1, torch.cuda.device_count())
+TASK_NUM_LABELS = {"boolq": 2, "rte": 2}
+DEFAULT_TASK = "rte"
+
+# narrow our logging to adapt it for a notebook environment
+for l_key in logging.Logger.manager.loggerDict.keys():
+    if "pytorch_lightning" in l_key:
+        logging.getLogger(l_key).setLevel("INFO")
+    else:
+        logging.getLogger(l_key).setLevel("CRITICAL")
+pl_logger = logging.getLogger("pytorch_lightning")
+pl_logger.removeHandler(pl_logger.handlers[0])
+rz_logger = logging.getLogger("pytorch_lightning.utilities.distributed")
+rz_logger.addHandler(logging.StreamHandler())
+rz_logger.handlers[0].setLevel("INFO")
+
+
+# %%
+class RteBoolqDataModule(pl.LightningDataModule):
+    """A ``LightningDataModule`` for using either the RTE or BoolQ SuperGLUE Hugging Face datasets."""
+
+    task_text_field_map = {"rte": ["premise", "hypothesis"], "boolq": ["question", "passage"]}
+    loader_columns = [
+        "datasets_idx",
+        "input_ids",
+        "token_type_ids",
+        "attention_mask",
+        "start_positions",
+        "end_positions",
+        "labels",
+    ]
+    # ignore warnings related tokenizers_parallelism/DataLoader parallelism tradeoff and
+    #  expected logging behavior
+    for warnf in [".*does not have many workers*", ".*The number of training samples.*"]:
+        warnings.filterwarnings("ignore", warnf)
+
+    def __init__(
+        self,
+        model_name_or_path: str,
+        task_name: str = DEFAULT_TASK,
+        prep_on_init: bool = False,
+        max_seq_length: int = 128,
+        train_batch_size: int = 32,
+        eval_batch_size: int = 32,
+        pin_memory: bool = False,
+        tokenizers_parallelism: bool = True,
+        num_workers: int = 0,
+    ):
+        super().__init__()
+        self.model_name_or_path = model_name_or_path
+        self.task_name = task_name if task_name in TASK_NUM_LABELS.keys() else DEFAULT_TASK
+        self.max_seq_length = max_seq_length
+        self.train_batch_size = train_batch_size
+        self.eval_batch_size = eval_batch_size
+        self.tokenizers_parallelism = tokenizers_parallelism
+        self.dataloader_kwargs = {"num_workers": num_workers, "pin_memory": pin_memory}
+        self.text_fields = self.task_text_field_map[self.task_name]
+        self.num_labels = TASK_NUM_LABELS[self.task_name]
+        os.environ["TOKENIZERS_PARALLELISM"] = "true" if self.tokenizers_parallelism else "false"
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name_or_path, use_fast=True, local_files_only=False)
+        if prep_on_init:  # useful if one wants to load datasets as soon as the ``LightningDataModule`` is
+            # instantiated
+            self.prepare_data()
```

Comment:
This isn't good practice. Let's remove it.

Reply:
Good find, I added this a long time ago and was blind to it.